sklearn/manifold/tests/test_t_sne.py

```diff
@@ -640,22 +640,23 @@ def test_barnes_hut_angle():
 def test_n_iter_without_progress():
     # Use a dummy negative n_iter_without_progress and check output on stdout
     random_state = check_random_state(0)
-    X = random_state.randn(100, 2)
-    tsne = TSNE(n_iter_without_progress=-1, verbose=2, learning_rate=1e8,
-                random_state=1, method='exact', n_iter=300)
-
-    old_stdout = sys.stdout
-    sys.stdout = StringIO()
-    try:
-        tsne.fit_transform(X)
-    finally:
-        out = sys.stdout.getvalue()
-        sys.stdout.close()
-        sys.stdout = old_stdout
-
-    # The output needs to contain the value of n_iter_without_progress
-    assert_in("did not make any progress during the "
-              "last -1 episodes. Finished.", out)
+    X = random_state.randn(100, 10)
+    for method in ["barnes_hut", "exact"]:
+        tsne = TSNE(n_iter_without_progress=-1, verbose=2, learning_rate=1e8,
```

Comment:
We changed the behavior of both `method`. Previously:
- For `method=barnes_hut`, this parameter was overridden to `30` for the whole optimization.
- For `method=exact`, this parameter was set to `self.n_iter_without_progress` for the whole optimization. And `n_iter_without_progress=-1`, mean that we stopped as soon as the optimization cost went up.

In the current implementation, we wanted to have a consistent optimization schedule for both method, so we implemented: 
- 250 iterations with `early_exaggeration` on and `n_iter_without_progress=250`
- Then we turn off `early_exaggeration` and use `self.n_iter_without_progress`.

Note that this parameter behavior also depends on the hard coded `n_iter_check=50` parameter set in `TSNE._tsne`. Thus `n_iter_without_progress < 50` is equivalent to 50, between 50/100, it is equivalent to 100. 
`n_iter_check` is implemented because it is expensive to compute the cost value but this value is currently computed at each iteration anyway. (This is one way to go to optimize the runtime)

I don't really know what is the best way to handle it. The failing test is a one that previously was used with `method=exact`. This test succeed as when `early_exaggeration` is on, with a large enough learning rate, this test passed.